Sources/TOONDecoder/TOONDecoder.swift

```diff
@@ -0,0 +1,1791 @@
+import Foundation
+
+/// A decoder that converts TOON format data to Swift values
+///
+/// This decoder conforms to the TOON (Token-Oriented Object Notation) specification version 3.0.
+/// For more information, see: https://github.com/toon-format/spec
+public final class TOONDecoder {
+    /// The TOON specification version this decoder conforms to
+    public static let specVersion = "3.0"
+
+    /// Number of spaces per indentation level
+    public var indent: Int = 2
```

Comment:
Should indentation be configurable? Or is this something that can be automatically determined by the parser?